### PR TITLE
TS0601 Dimmer gpp

### LINKED ISSUE #1103: [Device Support Request] Support for Tuya TS0601 dimmer
Hi, can you please add support for Tuya TS0601 dimmer?

Here is the divice description:

{
"node_descriptor": "NodeDescriptor(logical_type=<LogicalType.Router: 1>, complex_descriptor_available=0, user_descriptor_available=0, reserved=0, aps_flags=0, frequency_band=<FrequencyBand.Freq2400MHz: 8>, mac_capability_flags=<MACCapabilityFlags.AllocateAddress|RxOnWhenIdle|MainsPowered|FullFunctionDevice: 142>, manufacturer_code=4417, maximum_buffer_size=66, maximum_incoming_transfer_size=66, server_mask=10752, maximum_outgoing_transfer_size=66, descriptor_capability_field=<DescriptorCapability.NONE: 0>, *allocate_address=True, *is_alternate_pan_coordinator=False, *is_coordinator=False, *is_end_device=False, *is_full_function_device=True, *is_mains_powered=True, *is_receiver_on_when_idle=True, *is_router=True, *is_security_capable=False)",
"endpoints": {
"1": {
"profile_id": 260,
"device_type": "0x0051",
"in_clusters": [
"0x0000",
"0x0004",
"0x0005",
"0xef00"
],
"out_clusters": [
"0x000a",
"0x0019"
]
},
"242": {
"profile_id": 41440,
"device_type": "0x0061",
"in_clusters": [],
"out_clusters": [
"0x0021"
]
}
},
"manufacturer": "_TZE200_3p5ydos3",
"model": "TS0601",
"class": "zigpy.device.Device"
}

Thank you.
Tomas

### PR DESCRIPTION
**Pull request by MattWestb**
Adding GPP cluster and device "_TZE200_3p5ydos3".
Is getting status from the device but cant controlling it from ZHA.

Making is as draft so user can using it if they like but dont like merging in ZHA until it have more core function working.

Fixes #1103

### CODE REVIEW

**Comment by Tropaion** [comment]
@MattWestb Are you still working on this? Is there any progress?
I bought this and didn't know it isn't supported currently.

**Comment by MattWestb** [comment]
Not my but one very good user is working on tuya Dp dimmers and is reworking large things for getting it working OK https://github.com/zigpy/zha-device-handlers/pull/1247.
If you like try it out the work is don (not all os working but the ground is ther) look in https://github.com/zigpy/zha-device-handlers/issues/1033.

I was only making this for getting the first gang working for user that liking getting some function but is never going in and i deleting it the the new quirk is working and is being merged.

**Comment by Tropaion** [comment]
@MattWestb Thanks^^ I dont understand it completely because I'm pretty new in this field, but when will this be available/how long does it take that it gets to ZHA?

File: zhaquirks/tuya/ts0601_dimmer.py [hunk 1]
```diff
@@ -70,3 +78,64 @@ class TuyaSingleSwitchDimmer(TuyaDimmerSwitch):
...
+
+
+class TuyaSingleSwitchDimmerGP(TuyaDimmerSwitch):
+    """Tuya touch switch device."""
+
+    signature = {
+        # "node_descriptor": "<NodeDescriptor byte1=1, byte2=64, mac_capability_flags=142, manufacturer_code=4098,
+        # maximum_buffer_size=82, maximum_incoming_transfer_size=82, server_mask=11264,
+        # maximum_outgoing_transfer_size=82, descriptor_capability_field=0>",
+        # <SimpleDescriptor endpoint=1 profile=260 device_type=51 device_version=1 input_clusters=[0, 4, 5, 61184] output_clusters=[10, 25]>
```
Comment (javicalle) [inline zhaquirks/tuya/ts0601_dimmer.py:90]: The descriptor is not correct. I think it is better to remove it.

**Review by javicalle — CHANGES_REQUESTED** [review]

**Comment by MattWestb** [comment]
Thanks for comments @javicalle but i cant testing the device and the original user was not interested so i was "leaving it as WIP" for user can downloading it and getting the first gang working.
I think its better you is integrating this device in your work and can testing it with interested users.

Mvh  Mattias

**Comment by MattWestb** [comment]
Use the linked issues for implanting new devices with the new approach linked abow.

**Pull request closed without merging**